core/src/main/java/com/linecorp/armeria/common/util/UnmodifiableFuture.java

```diff
@@ -125,7 +125,7 @@ public boolean completeExceptionally(Throwable ex) {
         throw new UnsupportedOperationException();
     }
 
-    private void doCompleteExceptionally(Throwable cause) {
+    protected final void doCompleteExceptionally(Throwable cause) {
```

Comment:
- How about removing `final` so that a subclass can call this method?
- Javadoc